spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/inbound/ReactiveMongoDbMessageSource.java

```diff
@@ -0,0 +1,217 @@
+	 */
+	public void setEntityClass(Class<?> entityClass) {
+		Assert.notNull(entityClass, "'entityClass' must not be null");
+		this.entityClass = entityClass;
+	}
+
+	/**
+	 * Allows you to manage which find* method to invoke on {@link ReactiveMongoTemplate}.
+	 * Default is 'false', which means the {@link #receive()} method will use
+	 * the {@link ReactiveMongoTemplate#find(Query, Class)} method. If set to 'true',
+	 * {@link #receive()} will use {@link ReactiveMongoTemplate#findOne(Query, Class)},
+	 * and the payload of the returned {@link org.springframework.messaging.Message}
+	 * will be the returned target Object of type
+	 * identified by {{@link #entityClass} instead of a List.
+	 * @param expectSingleResult true if a single result is expected.
+	 */
+	public void setExpectSingleResult(boolean expectSingleResult) {
+		this.expectSingleResult = expectSingleResult;
+	}
+
+	/**
+	 * Sets the SpEL {@link Expression} that should resolve to a collection name
+	 * used by the {@link Query}. The resulting collection name will be included
+	 * in the {@link MongoHeaders#COLLECTION_NAME} header.
+	 * @param collectionNameExpression The collection name expression.
+	 */
+	public void setCollectionNameExpression(Expression collectionNameExpression) {
+		Assert.notNull(collectionNameExpression, "'collectionNameExpression' must not be null");
+		this.collectionNameExpression = collectionNameExpression;
+	}
+
+	/**
+	 * Allows you to provide a custom {@link MongoConverter} used to assist in deserialization
+	 * data read from MongoDb. Only allowed if this instance was constructed with a
+	 * {@link ReactiveMongoDatabaseFactory}.
+	 * @param mongoConverter The mongo converter.
+	 */
+	public void setMongoConverter(MongoConverter mongoConverter) {
+		Assert.isNull(this.reactiveMongoTemplate,
+				"'mongoConverter' can not be set when instance was constructed with ReactiveMongoTemplate");
+		this.mongoConverter = mongoConverter;
+	}
+
+	@Override
+	public String getComponentType() {
+		return "mongo:reactive-inbound-channel-adapter";
+	}
+
+	@Override
+	protected void onInit() {
+		this.evaluationContext =
+				ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
+		TypeLocator typeLocator = this.evaluationContext.getTypeLocator();
+		if (typeLocator instanceof StandardTypeLocator) {
+			//Register MongoDB query API package so FQCN can be avoided in query-expression.
+			((StandardTypeLocator) typeLocator).registerImport("org.springframework.data.mongodb.core.query");
+		}
+		if (this.reactiveMongoTemplate == null) {
+			this.reactiveMongoTemplate = new ReactiveMongoTemplate(this.reactiveMongoDatabaseFactory, this.mongoConverter);
+		}
+		this.initialized = true;
+	}
+
+	/**
+	 * Will execute a {@link Query} returning its results as the Message payload.
+	 * The payload can be either {@link Flux} of {@link Mono} of objects of type
+	 * identified by {{@link #entityClass}, or a single element of type identified by {{@link #entityClass}
```

Comment:
You have a lot of typos with double `{{` in this JavaDoc.

Reply:
Also in the original

Reply:
Yeah... So, looks like it would be great to fix all those flaws immediately when we see them. So, no one is going to be confused 😄